### PR TITLE
feat: add mongoose config

### PR DESCRIPTION
**Pull request by Sofiag8**
## Issue ticket number and link

https://trello.com/c/Rjpv241H/327-agregar-configuración-mongoose

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [ ] If it is a core feature, I have added thorough tests.
- [ ] Do we need to implement analytics?
- [ ] Will this be part of a product update? If yes, please write one phrase about this update.

## Pull request type

Please check the type of change your PR introduces:

- [x] :sparkles: Feature
- [ ] :bug: Bugfix
- [ ] :pencil2: Fixing typos
- [ ] :lipstick: :art: Code style update (formatting, renaming)
- [ ] :recycle: Refactoring (no functional changes, no api changes)
- [ ] :zap: Performance
- [ ] :pencil: Documentation content changes
- [ ] :white_check_mark: Unit testing
- [ ] :ok_hand: Updating due to code review suggest
- [ ] :rocket: Release: v
- [ ] :wrench: Add or update configuration files
- [ ] :hammer: Add or update development scripts
- [ ] :bricks: Infrastructure related changes

## What is the current module?

Right now we are not using mongoose

## What is the new behavior?

Add @nestjs/mongoose and mongoose packages, set mongoose connection config to app module, @nestjs/testing for future environment setup and update star:dev script due to previous changes 

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/app.module.ts [hunk 1]
```diff
@@ -1,8 +1,11 @@
 import { Module } from '@nestjs/common';
+import { ConfigModule } from '@nestjs/config';
+import { MongooseModule } from '@nestjs/mongoose';
+
 import { HealthController } from './modules/health-service/health.controller';
 
 @Module({
-  imports: [],
+  imports: [ ConfigModule.forRoot(), MongooseModule.forRoot(process.env.MONGODB_URI as string)],
```
Comment (FelipeLosada1996) [inline src/app.module.ts:8]: Create a file with variables, to utilize all you need, and call just the variable. https://www.linkedin.com/advice/1/how-do-you-keep-microservice-configurations-simple#:~:text=Centralized%20configuration%20allows%20you%20to,access%20control%20and%20security%20measures.

Comment (Sofiag8) [reply src/app.module.ts]: done

File: src/configuration/credentials.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+export default () => ({
+  mongo: {
+    host: process.env.DB_HOST,
```
[the remark this answers is not in the document]
Comment (Sofiag8) [reply src/configuration/credentials.ts]: done

**Review by FelipeLosada1996 — APPROVED** [review]

File: src/app.module.ts [hunk 3]
```diff
@@ -1,8 +1,19 @@
...
 
 @Module({
-  imports: [],
+  imports: [ 
+    ConfigModule.forRoot({
+      envFilePath: '.env', 
+      load: [credentials], 
+      isGlobal:true 
+    }), 
+    MongooseModule.forRoot(`mongodb://${credentials().MONGO.USER}:${credentials().MONGO.PASSWORD}@${credentials().MONGO.HOST}:${credentials().MONGO.PORT}/?authMechanism=DEFAULT`)
```
Comment (jose-sal-y-rosas) [inline src/app.module.ts:15]: Are you calling 3 times the function: credentials?

File: src/configuration/credentials.ts [hunk 4]
```diff
@@ -0,0 +1,9 @@
+export default () => ({
+  MONGO: {
```
Comment (jose-sal-y-rosas) [inline src/configuration/credentials.ts:2]: there is a specific file to handle environment variables, please could you check infrastructure/environment an try to use it?

**Review by jose-sal-y-rosas — CHANGES_REQUESTED** [review]
Please review changes

**Review by jose-sal-y-rosas — APPROVED** [review]
👍🏼

**Pull request merged**